lib/src/utils/serialization_utils.dart

```diff
@@ -923,3 +996,181 @@ class SerializationUtils {
+        StakeDelegation() => () {
+            final credential = cert.stakeCredential;
+            final void Function() invoker = switch (credential) {
+              CredentialKeyPath() => () => witnessPaths.add(credential.path),
+              _ => () => (),
+            };
+            invoker();
+          },
+        VoteDelegation() => () {
+            final credential = cert.stakeCredential;
+            final void Function() invoker = switch (credential) {
+              CredentialKeyPath() => () => witnessPaths.add(credential.path),
+              _ => () => (),
+            };
+            invoker();
+          },
+        AuthorizeCommitteeHot() => () {
+            final credential = cert.coldCredential;
+            final void Function() invoker = switch (credential) {
+              CredentialKeyPath() => () => witnessPaths.add(credential.path),
+              _ => () => (),
+            };
+            invoker();
+          },
+        ResignCommitteeCold() => () {
+            final credential = cert.coldCredential;
+            final void Function() invoker = switch (credential) {
+              CredentialKeyPath() => () => witnessPaths.add(credential.path),
+              _ => () => (),
+            };
+            invoker();
+          },
+        DRepRegistration() => () {
+            final credential = cert.dRepCredential;
+            final void Function() invoker = switch (credential) {
+              CredentialKeyPath() => () => witnessPaths.add(credential.path),
+              _ => () => (),
+            };
+            invoker();
+          },
+        DRepDeregistration() => () {
+            final credential = cert.dRepCredential;
+            final void Function() invoker = switch (credential) {
+              CredentialKeyPath() => () => witnessPaths.add(credential.path),
+              _ => () => (),
+            };
+            invoker();
+          },
+        DRepUpdate() => () {
+            final credential = cert.dRepCredential;
+            final void Function() invoker = switch (credential) {
+              CredentialKeyPath() => () => witnessPaths.add(credential.path),
+              _ => () => (),
+            };
+            invoker();
+          },
+        StakePoolRegistration() => () {
+            for (var owner in cert.pool.owners) {
+              final void Function() invoker = switch (owner) {
+                DeviceOwnedPoolOwner() => () => witnessPaths.add(owner.path),
+                _ => () => (),
+              };
+              invoker();
+            }
+            final poolKey = cert.pool.poolKey;
+            final void Function() invoker = switch (poolKey) {
+              DeviceOwnedPoolKey() => () => witnessPaths.add(poolKey.path),
+              _ => () => (),
+            };
+            invoker();
+          },
+        StakePoolRetirement() => () => witnessPaths.add(cert.path),
+        Object() => throw UnimplementedError(),
+        null => throw UnimplementedError(),
```

Comment:
This should not be needed. Indicates an issue somewhere.

Reply:
done